cli/src/main/java/org/eclipse/hono/cli/app/ReceiverStatistics.java

```diff
@@ -0,0 +1,218 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.cli.app;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.util.MessageHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * A command line client for statistic of received messages from via Hono's north bound Telemetry and/or Event API and
+ * <p>
+ * Messages are output to stdout.
+ * <p>
+ */
+@Component
+@Profile("statistic")
```

Comment:
Concerning the spring profile usage:
I would have expected to use `--spring.profiles.active=receiver,statistic` here, having the `statistic` profile as kind of a modifier. Having to run `--spring.profiles.active=statistic` (without `receiver`) feels unintuitive. And having to define extra properties for the `statistic` profile (same way as for the `receiver` profile) in the application.yml also doesn't look like a good solution.

How about having the `ReceiverStatistics` class not extend `Receiver`. Instead, the `Receiver` class could offer a `addMessageHandler` method, with all the added handlers getting invoked in the `handleMessage` method.
Then `ReceiverStatistics` could just be a bean that adds a handler via that `addMessageHandler` method, and the counting logic could be in that handler.

Reply:
Though I'm not to common with spring (or what ever technique is used for that), that seems to be out of my possibilities ;-).

Reply:
Just as additional idea:
Create a common abstract base class, derive the receiver with logging and the statistic from that. Would that be also OK?